public/app/features/explore/extensions/getExploreExtensionConfigs.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import { PluginExtensionPoints, type PluginExtensionLinkConfig } from '@grafana/data';
+import { contextSrv } from 'app/core/core';
+import { AccessControlAction, ExploreId } from 'app/types';
+
+import { createExtensionLinkConfig, logWarning } from '../../plugins/extensions/utils';
+
+import { AddToDashboardModal } from './AddToDashboard/AddToDashboardModal';
+import { type PluginExtensionExploreContext } from './ToolbarExtensionPoint';
+
+export function getExploreExtensionConfigs(): PluginExtensionLinkConfig[] {
+  try {
+    return [
+      createExtensionLinkConfig<PluginExtensionExploreContext>({
+        title: 'Dashboard',
+        description: 'Use the query and panel from explore and create/add it to a dashboard',
+        extensionPointId: PluginExtensionPoints.ExploreToolbarAction,
+        icon: 'apps',
+        configure: () => {
+          const canAddPanelToDashboard =
+            contextSrv.hasAccess(AccessControlAction.DashboardsCreate, contextSrv.isEditor) ||
+            contextSrv.hasAccess(AccessControlAction.DashboardsWrite, contextSrv.isEditor);
+
+          // hide option if user has insufficient permissions
+          if (!canAddPanelToDashboard) {
+            return undefined;
+          }
+
+          return {};
+        },
+        onClick: (_, { context, openModal }) => {
+          // temporary solution - will probably map values into context.
+          const exploreId = toExploreId(context?.exploreId);
+
+          openModal({
+            title: 'Add panel to dashboard',
+            body: ({ onDismiss }) => (
+              <AddToDashboardModal excludeModal={true} onClose={onDismiss!} exploreId={exploreId} />
```

Comment:
Just out of curiosity, why does TS need the `!` for `onDismiss`?

Reply:
Because the `AddToDashboardModal` doesn't accept a `() => void | undefined` only a `() => void`.